bin/eslint.js

```diff
@@ -45,33 +45,72 @@ function buildIssueJson(message, path) {
   return JSON.stringify(issue);
 }
 
-// Uses glob to traverse code directory and find files to analyze,
-// excluding files passed in with by CLI config, and including only
-// files in the list of desired extensions
-function fileWalk(excludePaths, extensions){
-  var analysisFiles = [];
-  var allFiles = glob.sync("/code/**/**", {});
-
-  allFiles.forEach(function(file, i, a){
-    if(excludePaths.indexOf(file.split("/code/")[1]) < 0) {
-      if(!fs.lstatSync(file).isDirectory()) {
-        var extension = "." + file.split(".").pop();
-
-        if(extensions.indexOf(extension) >= 0) {
-          analysisFiles.push(file);
+function exclusionBasedFileListBuilder(excludePaths) {
+  // Uses glob to traverse code directory and find files to analyze,
+  // excluding files passed in with by CLI config, and including only
+  // files in the list of desired extensions.
+  //
+  // Deprecated style of file expansion, supported for users of the old CLI.
+  return function(extensions) {
+    var analysisFiles = [];
+    var allFiles = glob.sync("/code/**/**", {});
```

Comment:
I think that's why it's in the `exclusionBasedFileListBuilder`, this is the deprecated, backwards-compatible path so it would still retain the bug -- as long as `inclusionBasedFileListBuilder` doesn't.